🐛 `lint/correctness/noUnusedVariables` quick fix breaks

### Environment information

```block
rome 11.0.0
```


### What happened?

```
let image: Buffer;

					try {
						// get typeid plus model as priority.
						image = SeedCore.getResource(
							p.typeID.toString() + '-' + p.model + '.jpg'
						);
					} catch (error) {
						try {
							image = SeedCore.getResource(p.typeID.toString() + '.jpg');
						} catch (error) {
							// do nothing
						}
					}
```

becomes

```
let image: Buffer;

					try {
						// get typeid plus model as priority.
						image = SeedCore.getResource(
							p.typeID.toString() + '-' + p.model + '.jpg'
						);
					} catch (error) {
						try {
							_error;
							image = SeedCore.getResource(p.typeID.toString() + '.jpg');
						} catch (error) {
							// do no_error
						}
					}
```

- syntax error (where the inserted _error; line is)
- doesn't actually fix the problem

### Expected result

should just work

### Code of Conduct

- [X] I agree to follow Rome's Code of Conduct

I have another example where the rule `noUnusedVariables` fails to detect an used variable and by providing a fix, which causes the variable to be renamed breaks the compilation.

With this snippet:

```typescript
export class Foo {
  constructor(readonly config: {foo: string}) {}

  bar() {
    return this.config.foo;
  }
}
```
I get the following output:

```
example.ts:2:24 lint/correctness/noUnusedVariables  FIXABLE  ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  ✖ This parameter is unused.
  
    1 │ export class Foo {
  > 2 │   constructor(readonly config: {foo: string}) {}
      │                        ^^^^^^
    3 │ 
    4 │   bar() {
  
  ℹ Unused variables usually are result of incomplete refactoring, typos and other source of bugs.
  
  ℹ Suggested fix: If this is intentional, prepend config with an underscore.
  
    1 1 │   export class Foo {
    2   │ - ··constructor(readonly·config:·{foo:·string})·{}
      2 │ + ··constructor(readonly·_config:·{foo:·string})·{}
    3 3 │   
    4 4 │     bar() {
  

Checked 1 file(s) in 411µs
Found 1 error(s)
Error: some errors were emitted while running checks
```

When applying the fix (`--apply-suggested`) we end up with:

```typescript
export class Foo {
  constructor(readonly _config: {foo: string}) {}

  bar() {
    return this.config.foo; // <-- compilation error as `config` isn't defined in the class
  }
}
```

`tsc` output is:

```
example.ts:5:17 - error TS2551: Property 'config' does not exist on type 'Foo'. Did you mean '_config'?

5     return this.config.foo;
                  ~~~~~~

  example.ts:2:15
    2   constructor(readonly _config: {foo: string}) {}
                    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    '_config' is declared here.
```

--- a/crates/rome_js_analyze/src/semantic_analyzers/correctness/no_unused_variables.rs
+++ b/crates/rome_js_analyze/src/semantic_analyzers/correctness/no_unused_variables.rs
@@ -7,7 +7,7 @@ use rome_js_semantic::{ReferencesExtensions, SemanticScopeExtensions};
 use rome_js_syntax::{
     binding_ext::{AnyJsBindingDeclaration, AnyJsIdentifierBinding, JsAnyParameterParentFunction},
     JsClassExpression, JsFunctionDeclaration, JsFunctionExpression, JsSyntaxKind, JsSyntaxNode,
-    JsVariableDeclarator, TsPropertyParameter,
+    JsVariableDeclarator,
 };
 use rome_rowan::{AstNode, BatchMutationExt};
 
@@ -136,20 +136,6 @@ fn is_function_that_is_ok_parameter_not_be_used(
     )
 }
 
-fn is_property_parameter_ok_not_be_used(parameter: TsPropertyParameter) -> Option<bool> {
-    for modifier in parameter.modifiers() {
-        if let Some(modifier) = modifier.as_ts_accessibility_modifier() {
-            match modifier.modifier_token().ok()?.kind() {
-                // modifiers that are ok to not be used
-                JsSyntaxKind::PRIVATE_KW | JsSyntaxKind::PUBLIC_KW => return Some(true),
-                _ => {}
-            }
-        }
-    }
-
-    Some(false)
-}
-
 fn is_ambient_context(node: &JsSyntaxNode) -> bool {
     node.ancestors()
         .any(|x| x.kind() == JsSyntaxKind::TS_DECLARE_STATEMENT)
@@ -174,16 +160,7 @@ fn suggested_fix_if_unused(binding: &AnyJsIdentifierBinding) -> Option<Suggested
         | AnyJsBindingDeclaration::JsFunctionExpression(_) => None,
 
         // Some parameters are ok to not be used
-        AnyJsBindingDeclaration::TsPropertyParameter(parameter) => {
-            let is_binding_ok =
-                is_function_that_is_ok_parameter_not_be_used(parameter.parent_function())
-                    || is_property_parameter_ok_not_be_used(parameter)?;
-            if !is_binding_ok {
-                suggestion_for_binding(binding)
-            } else {
-                None
-            }
-        }
+        AnyJsBindingDeclaration::TsPropertyParameter(_) => None,
         AnyJsBindingDeclaration::JsFormalParameter(parameter) => {
             let is_binding_ok =
                 is_function_that_is_ok_parameter_not_be_used(parameter.parent_function());

--- a/crates/rome_js_analyze/tests/specs/correctness/noUnusedVariables/validClass.ts
+++ b/crates/rome_js_analyze/tests/specs/correctness/noUnusedVariables/validClass.ts
@@ -13,8 +13,8 @@ console.log(new A(1));
 // we never flag class expressions
 new (class B { })
 
-// a and b are actually properties
+// a, b, c, and d are instance properties (declared as property parameters)
 class C {
-    constructor(private a1, public b2) {}
+	constructor(private a, public b, protected c, readonly d) {}
 }
-console.log(new C(1, 2));
+console.log(new C(1, 2, 3, 4));

